src/libstd/sys/windows/net.rs

```diff
@@ -178,15 +183,21 @@ impl Socket {
             cvt(c::WSADuplicateSocketW(self.0,
                                             c::GetCurrentProcessId(),
                                             &mut info))?;
+            #[cfg(not(target_os = "uwp"))]
+            let socket_flags = c::WSA_FLAG_OVERLAPPED;
+            #[cfg(target_os = "uwp")]
+            let socket_flags = c::WSA_FLAG_OVERLAPPED | c::WSA_FLAG_NO_HANDLE_INHERIT;
+
             match c::WSASocketW(info.iAddressFamily,
                                 info.iSocketType,
                                 info.iProtocol,
                                 &mut info, 0,
-                                c::WSA_FLAG_OVERLAPPED) {
+                                socket_flags) {
                 c::INVALID_SOCKET => Err(last_error()),
                 n => Ok(Socket(n)),
             }
         }?;
+        #[cfg(not(target_os = "uwp"))]
```

Comment:
Instead of sprinkling `#[cfg]` on all callers of `set_no_inherit` it may make more sense to just make that function a noop on uwp perhaps?

Reply:
I'm fine with both variants, it would probably hurt the readability less indeed, I'll do the change